### PR TITLE
docs: document dataclass_transform behavior

### PR DESCRIPTION
**Pull request by ikonst**
Document behavior discussed in https://github.com/python/typing/discussions/1456.

### CODE REVIEW

File: docs/source/additional_features.rst [hunk 1]
```diff
@@ -94,16 +94,35 @@ do **not** work:
...
+
+Data Class Transforms
+*********************
+
+Mypy supports the :py:func:`~typing.dataclass_transform` decorator as described in
+`PEP 681 <https://www.python.org/dev/peps/pep-0681/#the-dataclass-transform-decorator>`_.
+
+.. note::
+
+    Pragmatically, such classes would appear to have the internal attribute :code:`__dataclass_fields__`
```
Comment (hauntsaninja) [inline docs/source/additional_features.rst:122]:
```suggestion
    Pragmatically, mypy will assume such classes have the internal attribute :code:`__dataclass_fields__`
```

File: docs/source/additional_features.rst [hunk 2]
```diff
@@ -94,16 +94,35 @@ do **not** work:
...
+    (even though they might lack it in runtime) and dataclass functions such as :py:func:`dataclasses.is_dataclass`
+    and :py:func:`dataclasses.fields` would appear to accept them as if they were dataclasses
```
Comment (hauntsaninja) [inline docs/source/additional_features.rst:124]:
```suggestion
    (even though they might lack it in runtime) and will assume functions such as :py:func:`dataclasses.is_dataclass`
    and :py:func:`dataclasses.fields` to treat them as if they were dataclasses
```

File: docs/source/additional_features.rst [hunk 3]
```diff
@@ -94,16 +94,35 @@ do **not** work:
...
+
+To have Mypy recognize a wrapper of :py:func:`dataclasses.dataclass <dataclasses.dataclass>`
+as a dataclass decorator, consider using the :py:func:`~typing.dataclass_transform` decorator:
+
+.. code-block:: python
+
+    from dataclasses import dataclass, Field
+    from typing import dataclass_transform
+
+    @dataclass_transform(fields=(Field,))
```
Comment (hauntsaninja) [inline docs/source/additional_features.rst:108]: field_specifiers

Comment (ikonst) [reply docs/source/additional_features.rst:108]:
```suggestion
    @dataclass_transform(field_specifiers=(Field,))
```

File: docs/source/additional_features.rst [hunk 4]
```diff
@@ -94,16 +94,35 @@ do **not** work:
...
+*********************
+
+Mypy supports the :py:func:`~typing.dataclass_transform` decorator as described in
+`PEP 681 <https://www.python.org/dev/peps/pep-0681/#the-dataclass-transform-decorator>`_.
+
+.. note::
+
+    Pragmatically, mypy will assume such classes have the internal attribute :code:`__dataclass_fields__`
+    (even though they might lack it in runtime) and will assume functions such as :py:func:`dataclasses.is_dataclass`
+    and :py:func:`dataclasses.fields` to treat them as if they were dataclasses
```
Comment (hauntsaninja) [inline docs/source/additional_features.rst:124]:
```suggestion
    and :py:func:`dataclasses.fields` treat them as if they were dataclasses
```

File: docs/source/additional_features.rst [hunk 5]
```diff
@@ -94,16 +94,35 @@ do **not** work:
...
+
+.. code-block:: python
+
+    from dataclasses import dataclass, Field
+    from typing import dataclass_transform
+
+    @dataclass_transform(field_specifiers=(Field,))
+    def my_dataclass(cls: type[T]) -> type[T]:
+      ...
+      return dataclass(cls)
```
Comment (ikonst) [inline docs/source/additional_features.rst:111]:
```suggestion
    from dataclasses import dataclass, Field
    from typing import TypeVar, dataclass_transform
    
    T = TypeVar('T')

    @dataclass_transform(field_specifiers=(Field,))
    def my_dataclass(cls: type[T]) -> type[T]:
        ...
        return dataclass(cls)
```

**Pull request merged**